File: 2021/day01.py
def part1(nums):
    print(sum(x < y for x, y in zip(nums, nums[1:])))


def part2(nums):
    print(sum(x < y for x, y in zip(nums, nums[3:])))


with open("input/day01.txt") as f:
    v = list(map(int, f.readlines()))
    part1(v)
    part2(v)

File: 2021/day01.go
package main

import (
	"aoc/utils"
	"fmt"
)

func countGreaters(nums []int, diff int) (res int) {
	for i := diff; i < len(nums); i++ {
		if nums[i] > nums[i-diff] {
			res++
		}
	}
	return
}

// Part1 of day1
func Part1(nums []int) {
	fmt.Println(countGreaters(nums, 1))
}

// Part2 of day1
func Part2(nums []int) {
	fmt.Println(countGreaters(nums, 3))
}

func main() {
	nums := utils.StringSliceToIntSliceOrDie(utils.ReadFileLinesOrDie("./input/day01.txt"))
	Part1(nums)
	Part2(nums)
}
